src/medgen2obo.pl

```diff
@@ -133,14 +136,39 @@
                     $tag = 'is_a:';
                 }
                 if ($rel eq 'mapped_to') {
-                    $tag = 'equivalent_to:';
+                    # TODO: change these to skos:exactMatch?
+                    $tag = 'equivalent_to:';  # This translates to owl:equivalentClass
+                    # $tag = 'xref:';  # want to get this to translate to skos:exactMatch, but got oboInOwl:hasDbXref instead
+                }
+                # Namespaces: Different ones based on if is a UMLS CUI (C#), a MedGen CUI Novel (CN#), or a MedGEn UID (#).
```

Comment:
## Duplicate namespaces: For both classes _and_ references?
Bugfix: For UMLS CUIs (e.g. starts with C then #s), we chose to do duplicate classes with namespaces UMLS and MedGen. However, I just now made it so that also all references (e.g. xrefs) are also duplicated, e.g. MedGen:1 maps to MedGen:2 and UMLS:2.

Now I have the following branching logic in 2 places: (i) for creating classes, and (ii) for creating references:
```perl
if ($v =~ /^CN\d+/) {
    # Create class or reference using namespace: MedGenCUI
} else {
    if ($v =~ /^C\d+/) {
        # Create class or reference using namespace: UMLS
    }
    # Create class or reference using namespace: MedGen
    # This block runs for both UIDs & CUIs
```

Reply:
@matentzn Are we sure about this approach?

I believe the result will be that this:
```owl
<!--1 class; 1 set of refs-->
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/UMLS_C0000744">
        <owl:equivalentClass rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0154251"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/mondo/x-Disease-or-Syndrome.obo#CHD"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0020597"/>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>

<!--1 axiom-->
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <owl:annotatedProperty rdf:resource="http://www.w3.org/2002/07/owl#equivalentClass"/>
        <owl:annotatedTarget rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0000744"/>
        <oboInOwl:source rdf:datatype="http://www.w3.org/2001/XMLSchema#string">MSH</oboInOwl:source>
    </owl:Axiom>

```

Turns into this:
```owl
<!--Class 1 of 2, with duplicates of every ref-->
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/UMLS_C0000744">
        <owl:equivalentClass rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <owl:equivalentClass rdf:resource="http://purl.obolibrary.org/obo/MedGen_C2931925"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0154251"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0154251"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/mondo/x-Disease-or-Syndrome.obo#CHD"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0020597"/>
            </owl:Restriction>
        </rdfs:subClassOf>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/mondo/x-Disease-or-Syndrome.obo#CHD"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0020597"/>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>

<!--Class 2 of 2, with duplicates of every ref-->
    <owl:Class rdf:about="http://purl.obolibrary.org/obo/MedGen_C0000744">
        <owl:equivalentClass rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <owl:equivalentClass rdf:resource="http://purl.obolibrary.org/obo/MedGen_C2931925"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0154251"/>
        <rdfs:subClassOf rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0154251"/>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/mondo/x-Disease-or-Syndrome.obo#CHD"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0020597"/>
            </owl:Restriction>
        </rdfs:subClassOf>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="http://purl.obolibrary.org/obo/mondo/x-Disease-or-Syndrome.obo#CHD"/>
                <owl:someValuesFrom rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0020597"/>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>

<!--4 axioms-->
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <owl:annotatedProperty rdf:resource="http://www.w3.org/2002/07/owl#equivalentClass"/>
        <owl:annotatedTarget rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0000744"/>
        <oboInOwl:source rdf:datatype="http://www.w3.org/2001/XMLSchema#string">MSH</oboInOwl:source>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/MedGen_C2931925"/>
        <owl:annotatedProperty rdf:resource="http://www.w3.org/2002/07/owl#equivalentClass"/>
        <owl:annotatedTarget rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0000744"/>
        <oboInOwl:source rdf:datatype="http://www.w3.org/2001/XMLSchema#string">MSH</oboInOwl:source>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/UMLS_C2931925"/>
        <owl:annotatedProperty rdf:resource="http://www.w3.org/2002/07/owl#equivalentClass"/>
        <owl:annotatedTarget rdf:resource="http://purl.obolibrary.org/obo/MedGen_C0000744"/>
        <oboInOwl:source rdf:datatype="http://www.w3.org/2001/XMLSchema#string">MSH</oboInOwl:source>
    </owl:Axiom>
    <owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/MedGen_C2931925"/>
        <owl:annotatedProperty rdf:resource="http://www.w3.org/2002/07/owl#equivalentClass"/>
        <owl:annotatedTarget rdf:resource="http://purl.obolibrary.org/obo/UMLS_C0000744"/>
        <oboInOwl:source rdf:datatype="http://www.w3.org/2001/XMLSchema#string">MSH</oboInOwl:source>
    </owl:Axiom>
```

Reply:
I spoke with Nico and it did not seem like these duplicate references were a concern. I'm not sure if that's because MedGen may be taking over the bulk of the ingest work (they may even be able to provide us with a robot template; or we might just do a simple transform from whatever they create). But I also think Nico generally considered this as a non-issue, so marking resolved.